src/js/components/Box/StyledBox.js

```diff
@@ -67,32 +67,68 @@ const basisStyle = css`
 // https://stackoverflow.com/questions/36247140/why-doesnt-flex-item-shrink-past-content-size
 // we assume we are in the context of a Box going the other direction
 // TODO: revisit this
-const directionStyle = (direction, theme) => {
+const directionStyle = (directionWrapper, theme) => {
+  let direction;
+  let isResponsive;
+  let isReverse;
+
+  if (
+    typeof directionWrapper === 'string' ||
+    directionWrapper instanceof String
+  ) {
+    direction = directionWrapper;
+  } else {
+    // DirectionWrapper is object now.
+    direction = directionWrapper.direction;
+    isResponsive = directionWrapper.responsive;
+    isReverse = directionWrapper.reverse;
```

Comment:
Could those be deconstructed more nicely with ES6? Something like...
```
const { direction, responsive: isResponsive .... } = directionWrapper;
```

Reply:
Done